cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java

```diff
@@ -0,0 +1,229 @@
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.ChangeStreamIterable;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.ListDatabasesIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.MongoIterable;
+import com.mongodb.connection.ClusterDescription;
+import org.apache.camel.util.function.Suppliers;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class SslAwareMongoClient implements MongoClient {
+    private static final Logger LOG = LoggerFactory.getLogger(SslAwareMongoClient.class);
+    private static final TrustManager[] trustAllCerts = new TrustManager[] {
+            new X509TrustManager() {
+                public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                    return null;
+                }
+
+                @Override
+                public void checkClientTrusted(X509Certificate[] arg0, String arg1)
+                        throws CertificateException {
+                }
+
+                @Override
+                public void checkServerTrusted(X509Certificate[] arg0, String arg1)
+                        throws CertificateException {
+                }
+            }
+    };
+    private final Supplier<MongoClient> wrappedMongoClient = Suppliers.memorize(new Supplier<MongoClient>() {
+        @Override
+        public MongoClient get() {
+            String credentials = username == null ? "" : username;
+
+            if (!credentials.equals("")) {
+                credentials += password == null ? "@" : ":" + password + "@";
+            }
+
+            MongoClientSettings settings = MongoClientSettings.builder()
+                    .applyToSslSettings(builder -> {
+                        builder.enabled(ssl);
+                        if (sslAllowInvalidCertificatesAndHostName) {
+                            builder.invalidHostNameAllowed(true);
+                            SSLContext sc = null;
+                            try {
+                                sc = SSLContext.getInstance("SSL");
+                            } catch (NoSuchAlgorithmException e) {
+                                e.printStackTrace();
+                            }
+                            try {
+                                sc.init(null, trustAllCerts, new java.security.SecureRandom());
+                            } catch (KeyManagementException e) {
+                                e.printStackTrace();
```

Comment:
should perhaps re-throw a runtime exception ?

Reply:
done